### PR TITLE
[PT FE] Added aten::logaddexp

### PR DESCRIPTION
**Pull request by itsbharatj**
### Details:

Added `aten::logcumsumexp`

### Tickets:
 - *ticket-id*:_
 - Issue ID: #28538

### CODE REVIEW

File: src/frontends/pytorch/src/op/logaddexp.cpp [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    // Add the exponentials
+    auto sum = context.mark_node(std::make_shared<v1::Add>(exp1, exp2));
+
+    // Take the natural logarithm of the sum
+    auto result = context.mark_node(std::make_shared<v0::Log>(sum));
+
+    // If the output tensor type is different, convert to match
+    if (input1.get_element_type() != element::f32) {
+        result = context.mark_node(std::make_shared<v1::ConvertLike>(result, input1));
+    }
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/logaddexp.cpp:45]: I think you provide numerically unstable decomposition for this operation, that means for too large inputs we can experience exponent explosion like for softmax.

That is why I propose:
1. to find maximum from two inputs k 
2. ln(e^x1 + e^x2) = ln ( e^k (e^(x1-k) + e^(x2-k))) = k + ln ( e^(x1-k) + e^(x2-k))

**Comment by rkazants** [comment]
build_jenkins

File: src/frontends/pytorch/src/op/logaddexp.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/logaddexp.cpp:1]:
```suggestion
// Copyright (C) 2018-2025 Intel Corporation
```

File: src/frontends/pytorch/src/op/logaddexp.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/add.hpp"
+#include "openvino/op/constant.hpp"
```
Comment (rkazants) [inline src/frontends/pytorch/src/op/logaddexp.cpp:7]:
```suggestion
```
not used

File: src/frontends/pytorch/src/op/logaddexp.cpp (hunk 2 above)
Comment (itsbharatj) [reply src/frontends/pytorch/src/op/logaddexp.cpp]: Thanks for pointing it out! 
This is my first time contributing, would make sure to keep an eye for these things from the next time

**Comment by itsbharatj** [comment]
Can you let me know for any changes required/approve the CI request.

File: tests/layer_tests/pytorch_tests/test_logaddexp.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+        class LogAddExpModel(torch.nn.Module):
+            def __init__(self, dtype=None):
+                super(LogAddExpModel, self).__init__()
+                self.dtype = dtype_map.get(dtype) if dtype else None
+
+            def forward(self, x, y):
+                if self.dtype:
+                    x = x.to(self.dtype)
+                    y = y.to(self.dtype)
```
Comment (rkazants) [inline tests/layer_tests/pytorch_tests/test_logaddexp.py:30]: I think we don't need this casting. It is sufficient to generate input data of required type.

Comment (rkazants) [reply tests/layer_tests/pytorch_tests/test_logaddexp.py:30]: please remove and the map as well

File: tests/layer_tests/pytorch_tests/test_logaddexp.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+                return torch.logaddexp(x, y)
+
+        model_class = LogAddExpModel(dtype)
+        ref_net = None
+
+        return model_class, ref_net, "aten::logaddexp"
+
+    @pytest.mark.nightly
+    @pytest.mark.precommit
+    @pytest.mark.parametrize(
```
Comment (rkazants) [inline tests/layer_tests/pytorch_tests/test_logaddexp.py:40]: please also add `@pytest.mark.precommit_torch_export`. Take a look how other operations are enabled for torch.export case.
In order to test it locally, set env var `PYTORCH_TRACING_MODE=EXPORT`: https://github.com/openvinotoolkit/openvino/blob/master/.github/workflows/job_pytorch_layer_tests.yml#L132

File: src/frontends/pytorch/src/op/logaddexp.cpp [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+OutputVector translate_logaddexp(const NodeContext& context) {
+    // "aten::logaddexp(Tensor self, Tensor other) -> Tensor"
+    num_inputs_check(context, 2, 2);
+
+    auto input1 = context.get_input(0);
+    auto input2 = context.get_input(1);
+
+    // Convert inputs to floating point type if needed
+    input1 = context.mark_node(std::make_shared<v0::Convert>(input1, element::f32));
+    input2 = context.mark_node(std::make_shared<v0::Convert>(input2, element::f32));
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/logaddexp.cpp:31]:
```suggestion
    Output<Node> input1;
    Output<Node> input2;
    std::tie(input1, input2) = get_inputs_with_promoted_types(context, 0, 1);
```
This will promote inputs to the common type, since `torch.logaddexp` require one of the inputs to be of floating type, the common type will be floating type.

File: src/frontends/pytorch/src/op/logaddexp.cpp [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+    // Take the log and add back the maximum
+    auto log_sum = context.mark_node(std::make_shared<v0::Log>(sum));
+    auto result = context.mark_node(std::make_shared<v1::Add>(log_sum, max_val));
+
+    // If the output tensor type is different, convert to match
+    if (input1.get_element_type() != element::f32) {
+        result = context.mark_node(std::make_shared<v1::ConvertLike>(result, input1));
+    }
+
```
Comment (mvafin) [inline src/frontends/pytorch/src/op/logaddexp.cpp:55]:
```suggestion
```
No need for this if we do calculations in the common dtype.

File: tests/layer_tests/pytorch_tests/test_logaddexp.py [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        "dtype",
+        [
+            "float32",
+            "float64",
```
Comment (mvafin) [inline tests/layer_tests/pytorch_tests/test_logaddexp.py:44]: Please add mixed dtypes case when one of the inputs is float and another is int

**Review by mvafin — CHANGES_REQUESTED** [review]
The tests fail for `FP64` dtype, because you align the output of the operation to `input1` which you previously converted to `FP32`. With my suggestions the tests should pass.

**Review by mvafin — CHANGES_REQUESTED** [review]
You haven't apply my comments

**Comment by itsbharatj** [comment]
@mvafin I have tried to resolve all the comments, please take a look and let me know for any further changes. 
Thanks

**Comment by p-wysocki** [comment]
@mvafin could you please take a look?

File: src/plugins/intel_cpu/thirdparty/onednn
Comment (mvafin) [inline src/plugins/intel_cpu/thirdparty/onednn:1]: why is this changed? Please revert

File: src/plugins/intel_gpu/thirdparty/onednn_gpu
Comment (mvafin) [inline src/plugins/intel_gpu/thirdparty/onednn_gpu:1]: Please revert this file

File: src/plugins/intel_npu/thirdparty/level-zero-ext
Comment (mvafin) [inline src/plugins/intel_npu/thirdparty/level-zero-ext:1]: Please revert this file

File: thirdparty/level_zero/level-zero
Comment (mvafin) [inline thirdparty/level_zero/level-zero:1]: Please revert this file

**Review by mvafin — CHANGES_REQUESTED** [review]
